### PR TITLE
#41 added more filters and fields to the table view

### PR DESCRIPTION
**Pull request by MartinMaritim**

### CODE REVIEW

File: sites/all/modules/hr/hr_datasets/hr_datasets.module [hunk 1]
```diff
@@ -139,18 +139,32 @@ function hr_datasets_block_view($delta = '') {
...
+      drupal_add_library('chosen', 'drupal.chosen');
       $block['content'] = '<div><div class="views-exposed-form">
         <div class="views-exposed-widgets clearfix">
                 <div id="edit-search-wrapper" class="views-exposed-widget form-group">
         <div class="input-group">
-          <input class="form-control form-text" id="search" name="datasets.title" value="" size="30" maxlength="128" type="text" placeholder="Search Datasets by Title">
-          <span class="input-group-btn">
-            <button id="search-button" type="submit" class="btn btn-default"><i class="fa fa-search"></i></button>
-          </span>
+          <input class="form-control text-center" id="search" name="datasets.title" value="" maxlength="128" type="text" placeholder="Search Datasets by Title" style="width: 255px">
```
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/hr_datasets.module]: Why do we have html forms in blocks?

File: sites/all/modules/hr/hr_datasets/hr_datasets.module [hunk 2]
```diff
@@ -139,18 +139,32 @@ function hr_datasets_block_view($delta = '') {
   $block = array();
   switch ($delta) {
     case 'hdx-datasets_search':
+      drupal_add_library('chosen', 'drupal.chosen');
```
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/hr_datasets.module]: It's better to use #attached when possible

File: sites/all/modules/hr/hr_datasets/hr_datasets.module [hunk 3]
```diff
@@ -165,5 +179,31 @@ break;
...
+*/
+
+function hr_datasets_hdx_tags() {
+  $url = "https://data.hdx.rwlabs.org/api/3/action/tag_list";
+  $request = drupal_http_request($url);
+  if(!empty($request->data)) {
+    $result = drupal_json_decode($request->data);
+    return $result['result'];
+  }
+}
```
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/hr_datasets.module]: This bit is identical in both functions, we should have a common way to read from HDX and use that utility function

File: sites/all/modules/hr/hr_datasets/js/hr_datasets.js [hunk 4]
```diff
@@ -142,12 +147,36 @@ Drupal.behaviors.hrDatasets = {
         },
 
         searchByTitle: function(event) {
-          var val = $('#search').val();
+          if(event.keyCode == 13) {
```
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/js/hr_datasets.js]: What is 13?

Comment (MartinMaritim) [inline sites/all/modules/hr/hr_datasets/js/hr_datasets.js]: This refers to the event when the 'Enter/Return' key is pressed.

File: sites/all/modules/hr/hr_datasets/hr_datasets.module (hunk 3 above)
Comment (MartinMaritim) [inline sites/all/modules/hr/hr_datasets/hr_datasets.module]: Noted. I will add an argument to the function, and make it more generic ...

Comment (MartinMaritim) [inline sites/all/modules/hr/hr_datasets/hr_datasets.module]: Update: This has been addressed here https://github.com/humanitarianresponse/site/commit/283be1a15decb37fbe14e0433cbac513440536ce

File: sites/all/modules/hr/hr_datasets/js/hr_datasets.js (hunk 4 above)
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/js/hr_datasets.js]: I think we need to use which as well as keycode then.
Does this work well on mobile?

File: sites/all/modules/hr/hr_datasets/js/hr_datasets.js [hunk 5]
```diff
@@ -142,12 +149,36 @@ Drupal.behaviors.hrDatasets = {
...
+            var val = $('#search').val();
+            if (val != '') {
+              this.DatasetsList.params.title = val;
```
Comment (pcambra) [inline sites/all/modules/hr/hr_datasets/js/hr_datasets.js]: you're doing this if val not empty add value, else remove value then navigate several times, I think we should have an helper function here.

**Comment by pcambra** [comment]
Why do we have a merge commit in there? could you squash them together?

**Pull request closed without merging**